### PR TITLE
[WIP] Add round robin 

### PR DESCRIPTION
**Pull request by dulvinw**
$title

### CODE REVIEW

**Comment by dulvinw** [comment]
@gihanlaks

File: README.md [hunk 1]
```diff
@@ -29,12 +29,22 @@ pip install mpi4py
...
-cd cpp
+cd cpp\build
 mpirun -n 4 ./hello_world
 ```
 
+```bash
+cd cpp\build
+mpirun -n 4 ./send_recv <program-num>
+
+Program Numbers:
```
Comment (vibhatha) [inline README.md:40]: Rather let's write with <program_num>

File: README.md [hunk 2]
```diff
@@ -29,12 +29,22 @@ pip install mpi4py
...
+1. Send Number 
+2. Send Array
+3. Send vector
```
Comment (vibhatha) [inline README.md]: vector or Vector

File: cpp/src/round_robin.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+void mpi_send_number(int number_to_send, int destination, int rank) {
+    cout << format("{}. Sending to rank {} ", rank, destination) << endl;
+    MPI_Send(&number_to_send, 1, MPI_INT, destination, 0, MPI_COMM_WORLD);
+    cout << format("{}. sent number {} to rank {}", rank, number_to_send, destination) << endl;
+}
+
+void mpi_receive(int *receive, int source, int rank) {
+    cout << format("{}. Receiving from rank {}", rank, source) << endl;
+    MPI_Recv(receive, 1, MPI_INT, 0, 0, MPI_COMM_WORLD, MPI_STATUS_IGNORE);
```
Comment (vibhatha) [inline cpp/src/round_robin.cpp]: Do we have to use `MPI_STATUS_IGNORE`?

File: cpp/src/send_recv.cpp [hunk 4]
```diff
@@ -41,16 +41,17 @@ int simple_send_recv() {
         return EXIT_FAILURE;
     }
 
-    int number;
+//    int number;
```
Comment (vibhatha) [inline cpp/src/send_recv.cpp]: Let's remove this comment

File: cpp/src/round_robin.cpp (hunk 3 above)
Comment (dulvinw) [reply cpp/src/round_robin.cpp]: I'm not sure. Copied from send_recv :)

File: .idea/.gitignore [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Editor-based HTTP Client requests
+/httpRequests/
+# Datasource local storage ignored files
+/dataSources/
+/dataSources.local.xml
```
Comment (vibhatha) [inline .idea/.gitignore]: do we need this file? I am not sure if this is coming from `git commit -am` ?

File: cpp/src/send_recv.cpp [hunk 6]
```diff
@@ -28,7 +28,7 @@ int main(int argc, char** argv) {
 }
 
 int simple_send_recv() {
-    MPI_Init(NULL, NULL);
```
Comment (vibhatha) [inline cpp/src/send_recv.cpp:31]: specific reason?

File: .idea/.gitignore (hunk 5 above)
Comment (dulvinw) [reply .idea/.gitignore]: No. Need to add this to gitignore

File: cpp/src/send_recv.cpp (hunk 6 above)
Comment (dulvinw) [reply cpp/src/send_recv.cpp:31]: Used `nullptr` below. Better typesafety :)

**Review by vibhatha — APPROVED** [review]
LGTM! Thanks @dulvinw 
@gihanlaks wanna take a look?

**Review by gihanlaks — APPROVED** [review]
This look okay to my cpp poor eyes

**Comment by vibhatha** [comment]
@gihanlaks you can merge this one when you're ready.

**Comment by gihanlaks** [comment]
merged

**Pull request merged**